Python/Product/PythonTools/PythonTools/Intellisense/CompletionComparer.cs

```diff
@@ -92,20 +92,31 @@ class CompletionComparer : IEqualityComparer<CompletionResult>, IComparer<Comple
         /// <param name="y"></param>
         /// <returns></returns>
         public int Compare(CompletionResult x, CompletionResult y) {
-            return Compare(x.Name, y.Name);
+            int i = Compare(x.Completion, y.Completion);
+            return i == 0 ? 0 : Compare(x.Name, y.Name);
         }
 
         /// <summary>
         /// Compares two <see cref="MemberResult"/> structures for equality.
         /// </summary>
         public bool Equals(CompletionResult x, CompletionResult y) {
-            return x.Name.Equals(y.Name);
+            return x.Name.Equals(y.Name) || x.Completion.Equals(y.Completion);
```

Comment:
This implementation of `Equals` breaks transitive relation:

|      | Name | Completion |
| --- | ------ | -------- |
| `x` | "ab" | "bc" |
| `y` | "ab" | "cd" |
| `z` | "de" | "cd" |

`completionComparer.Equals(x, y) == true`, `completionComparer.Equals(y, z) == true`, but `completionComparer.Equals(x, z) == false`,

Reply:
You're right, we're going to take a different approach. Hugues is working on it